src/7-new-content/4-data-formats.adoc

```diff
@@ -0,0 +1,188 @@
+= Formatting API data
+
+A request body (and a response body) will have a `Content-Type`, and that
+content type will tell tools how the data is formatted so it can be converted
+into something meaningful in whichever programming language is being used.
+
+== JSON: The Modern Standard
+
+JSON has become the de facto standard for API requests because it:
+
+* Supports native data types (numbers, booleans, null)
+* Allows nested structures
+* Is human-readable
+* Has excellent tooling support
+
+Example of complex JSON request:
+
+{lang=json}
+~~~~~~~~
+{
+  "place": {
+    "name": "Central Park",
+    "location": {
+      "lat": 40.785091,
+      "lon": -73.968285
+    },
+    "features": ["park", "landmark"],
+    "isAccessible": true,
+    "capacity": null
+  }
+}
+~~~~~~~~
+
+== XML: Ye Oldé Standard
+
+Any modern API you interact with will support JSON. Occasionally they will
+support XML as well.
+
+XML is relict of the early internet. It dominated web APIs in the 2000s with
+standards like SOAP and XML-RPC, but was largely displaced by JSON in the 2010s
+due to JSON's simplicity and natural fit with JavaScript. Today, XML persists
+mainly in legacy systems, enterprise SOAP services, and specific domains like
+publishing (DocBook), feed syndication (RSS/Atom), and configuration files
+(Maven, Android manifests).
+
+JSON is a lot easier to work with than XML, and it is a lot easier to read. It is
+also more compact, which is important when you are sending data over the wire.
+
+An example of a bunch of different data types in JSON.
+
+{lang=json}
+~~~~~~~~
+{
+  "place": {
+    "id": 1,
+    "name": "This is a bunch of text.",
+    "is_true": false,
+    "maybe": null,
+    "empty_string": ""
+  }
+}
+~~~~~~~~
+
+{lang=xml}
+~~~~~~~~
+<places>
+    <place>
+        <id>1</id>,
+        <name>This is a bunch of text.</name>
+        <is_true>0</is_true>
+        <maybe />
+        <empty_string />
+    </place>
+</places>
+~~~~~~~~
+
+Basically, in XML, _everything_ is considered a string, meaning integers,
+booleans, and nulls can be confused. Both `maybe` and `empty_string` have the
```

Comment:
Not sure what I’m saying with `maybe` and `empty_string` there, that seems confused after a few edits.